cloud/parameterstore/ssm_client.go

```diff
@@ -0,0 +1,178 @@
+	"github.com/evergreen-ci/utility"
+	"github.com/mitchellh/mapstructure"
+	"github.com/mongodb/grip"
+	"github.com/mongodb/grip/message"
+	"github.com/pkg/errors"
+	"go.opentelemetry.io/contrib/instrumentation/github.com/aws/aws-sdk-go-v2/otelaws"
+)
+
+type ssmClient interface {
+	// PutParameter puts a parameter into Parameter Store.
+	PutParameter(context.Context, *ssm.PutParameterInput) (*ssm.PutParameterOutput, error)
+	// DeleteParametersSimple is the same as DeleteParameters but only returns
+	// the names of the deleted parameters rather than the full output.
+	DeleteParametersSimple(context.Context, *ssm.DeleteParametersInput) ([]string, error)
+	// DeleteParameters deletes the specified parameters. Parameter Store limits
+	// how many parameters can be deleted per call, so implementations are
+	// expected to handle batching transparently. The returned output slice contains the
+	// output of each batched request.
+	DeleteParameters(context.Context, *ssm.DeleteParametersInput) ([]*ssm.DeleteParametersOutput, error)
+	// GetParametersSimple is the same as GetParameters but only returns the
+	// parameter information rather than the full output.
+	GetParametersSimple(context.Context, *ssm.GetParametersInput) ([]ssmTypes.Parameter, error)
+	// GetParameters retrieves the specified parameters. Parameter Store
+	// limits how many parameters can be retrieved per call, so implementations
+	// must handle batching transparently. The returned output slice contains
+	// the output of each batched request.
+	GetParameters(context.Context, *ssm.GetParametersInput) ([]*ssm.GetParametersOutput, error)
+}
+
+type ssmClientImpl struct {
+	client *ssm.Client
+}
+
+// configCache is a mapping of configuration options to the cached AWS
+// configuration created by it. Currently the only supported option is the AWS
+// region.
+var configCache map[string]*aws.Config = make(map[string]*aws.Config)
+
+func newSSMClient(ctx context.Context, region string) (ssmClient, error) {
+	if region == "" {
+		region = "us-east-1"
+	}
+
+	if configCache[region] == nil {
+		cfg, err := config.LoadDefaultConfig(ctx,
+			config.WithRegion(region),
+		)
+		if err != nil {
+			return nil, errors.Wrap(err, "loading AWS config")
+		}
+		otelaws.AppendMiddlewares(&cfg.APIOptions)
+		configCache[region] = &cfg
+	}
+
+	c := ssm.NewFromConfig(*configCache[region])
+
+	return &ssmClientImpl{client: c}, nil
+}
+
+func (c *ssmClientImpl) PutParameter(ctx context.Context, input *ssm.PutParameterInput) (*ssm.PutParameterOutput, error) {
+	return retrySSMClientOp(ctx, "PutParameter", c.client, input, func(ctx context.Context, client *ssm.Client, input *ssm.PutParameterInput) (*ssm.PutParameterOutput, error) {
+		return client.PutParameter(ctx, input)
+	})
+}
+
+func (c *ssmClientImpl) DeleteParametersSimple(ctx context.Context, input *ssm.DeleteParametersInput) ([]string, error) {
+	output, err := c.DeleteParameters(ctx, input)
```

Comment:
Does retrySSMClientOp belong here as well?

Reply:
It's calling `DeleteParameters` which already retries the op. This is just a little wrapper to get the most useful information (i.e. the deleted parameter names) out of it.